Sources/Fluent/Query/Join.swift

```diff
@@ -12,16 +12,41 @@
 public struct Join {
     /// Entity that will be accepting
     /// the joined data
-    public let local: Entity.Type
+    public let base: Entity.Type
 
     /// Entity that will be joining
-    /// the local data
-    public let foreign: Entity.Type
+    /// the base data
+    public let joined: Entity.Type
+
+    /// See Child enum
+    public let child: Child
+
+    /// Indicates which entity contains
+    /// a foreign id pointer to the other entity
+    public enum Child {
+        /// The base entity contains
+        /// a foreign id pointer to the joined data
+        ///
+        /// base        | joined
+        /// ------------+-------
+        /// joined_id   | id
+        case base
+        /// The joined entity contains
+        /// a foreign id pointer to the base data
+        /// 
+        /// base | joined
+        /// -----+--------
+        /// id   | base_id
+        ///
+        /// This is the default case.
+        case joined
+    }
 
     /// Create a new Join
-    public init(local: Entity.Type, foreign: Entity.Type) {
-        self.local = local
-        self.foreign = foreign
+    public init(base: Entity.Type, joined: Entity.Type, child: Child = .joined) {
+        self.base = base
+        self.joined = joined
+        self.child = child
```

Comment:
There's more than just the order involved. The `joined` table is what gets put in `JOIN foo_table`. If you switch the order of base and joined you'd have something like:

```sql
SELECT * FROM users JOIN users on foo.id = users.foo_id
```

This will error that `foo.id` could not be found (since the foo table was not ever joined)

What we actually want is the ability to differentiate between:

```sql
SELECT * FROM users JOIN foo ON users.id = foo.user_id
```
and 
```sql
SELECT * FROM users JOIN foo on users.foo_id = foo.id
```